src/main/fc/msp_server_fc.c

```diff
@@ -962,6 +972,28 @@ int mspServerCommandHandler(mspPacket_t *cmd, mspPacket_t *reply)
             serializeSDCardSummaryReply(reply);
             break;
 
+        case MSP_BATTERY_STATES:
+            sbufWriteU8(dst, (uint8_t)getBatteryState() == BATTERY_NOT_PRESENT ? 0 : 1); // battery connected - 0 not connected, 1 connected
+            sbufWriteU8(dst, (uint8_t)constrain(vbat, 0, 255));
+            sbufWriteU16(dst, (uint16_t)constrain(getCurrentMeter(batteryConfig()->currentMeterSource)->mAhDrawn, 0, 0xFFFF)); // milliamp hours drawn from battery
```

Comment:
temporary variable will help here. 

Also - is `getCurrentMeter` guarantied to return non-NULL pointer?